src/Migrate/deploymentscripts/Initialize-AzMigrateReplicationInfrastructure.ps1

```diff
@@ -0,0 +1,386 @@
+            $serviceBusConnString = "ServiceBusConnectionString"
+            $serviceBusSecretObject = Get-AzKeyVaultSecret -VaultName $KeyVaultName -Name $serviceBusConnString -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if($serviceBusSecretObject){
+                Write-Host $serviceBusConnString, " for ", $applianceName, $LogStringSkipping
+            }else{
+                $serviceBusRootKey = Get-AzServiceBusKey -ResourceGroupName $ResourceGroupName -Namespace $ServiceBusNamespace -Name "RootManageSharedAccessKey"
+                $secret = ConvertTo-SecureString -String $serviceBusRootKey.PrimaryConnectionString -AsPlainText -Force
+                Set-AzKeyVaultSecret -VaultName $KeyVaultName -Name $serviceBusConnString -SecretValue $secret
+                Write-Host $LogStringCreated, $serviceBusConnString, " for ", $applianceName
+            }  
+
+            # Policy
+            $policyName = $MigratePrefix + $SiteName + "policy"
+            $existingPolicyObject = Get-AzMigrateReplicationPolicy -PolicyName $policyName -ResourceGroupName $ResourceGroupName -ResourceName $VaultName -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if(!$existingPolicyObject){
+                $providerSpecificPolicy = [Microsoft.Azure.PowerShell.Cmdlets.Migrate.Models.Api20180110.VMwareCbtPolicyCreationInput]::new()
+                $providerSpecificPolicy.AppConsistentFrequencyInMinute = 240
+                $providerSpecificPolicy.InstanceType = "VMwareCbt"
+                $providerSpecificPolicy.RecoveryPointHistoryInMinute = 4320
+                $providerSpecificPolicy.CrashConsistentFrequencyInMinute = 60
+                $existingPolicyObject = New-AzMigrateReplicationPolicy -PolicyName $policyName -ResourceGroupName $ResourceGroupName -ResourceName $VaultName -ProviderSpecificInput $providerSpecificPolicy
+                Write-Host $LogStringCreated, $policyName
+            }else{
+                Write-Host $policyName, $LogStringSkipping
+            }
+
+            # Policy-container mapping
+            $mappingName = "containermapping"
+            $allFabrics = Get-AzMigrateReplicationFabric -ResourceGroupName $ResourceGroupName -ResourceName $VaultName
+            foreach($fabric in $allFabrics){
+                if($fabric.Name -match $applianceName){
+                    $peContainers = Get-AzMigrateReplicationProtectionContainer -FabricName $fabric.Name -ResourceGroupName $ResourceGroupName -ResourceName $VaultName
+                    foreach ($peContainer in $peContainers) {
+                        if($peContainer.Name -match $applianceName){
+                            $existingMapping = Get-AzMigrateReplicationProtectionContainerMapping -ResourceGroupName $ResourceGroupName -ResourceName $VaultName -FabricName $fabric.Name -ProtectionContainerName $peContainer.Name -MappingName $mappingName -ErrorVariable notPresent -ErrorAction SilentlyContinue
```

Comment:
you need this check before creating artifacts. if mapping exists and target location provided is same as existing mapping, then you can proceed else you need to block creating all artifacts.

in portal, once container mapping created, customer cannot change target location. please add same handling here as well.

Reply:
done, made a check at top too.